apps/model/model.js

```diff
@@ -994,3 +1072,281 @@ function downloadCSV(filename) {
     self.showResults('Please select a model first');
   }
 }
+
+async function selectModel() {
+  var data = await tf.io.listModels();
+  var table = document.querySelector('#roidata');
+  var tx = db.transaction('models_store', 'readonly');
+  var store = tx.objectStore('models_store');
+  var modelCount=0;
+  empty(table);
+
+  //  Update table data
+  (function(callback) {
+    for (const key in data) {
+      if (data.hasOwnProperty(key)) {
+        const name = key.split('/').pop();
+        const date = data[key].dateSaved.toString().slice(0, 15);
+        const size = (data[key].modelTopologyBytes +
+          data[key].weightDataBytes +
+          data[key].weightSpecsBytes) / (1024*1024);
+        const row = table.insertRow();
+        let classes; let inputShape; let td;
+
+        if (name.slice(0, 4) == 'pred') {
+          store.get(name).onsuccess = function(e) {
+            classes = (e.target.result.classes.join(', '));
+            inputShape = e.target.result.input_shape.slice(1, 3).join('x');
+            td = row.insertCell();
+            td.innerHTML = name.split('/').pop().split('_').splice(2).join('_').slice(0, -3);
```

Comment:
would innertext be safer?

Reply:
Yes that would be much better. I'll change it.